src/towncrier/_project.py

```diff
@@ -39,8 +41,10 @@ def _get_package(package_dir: str, package: str) -> ModuleType:
 
 def get_version(package_dir: str, package: str) -> str:
     module = _get_package(package_dir, package)
-
-    version = getattr(module, "__version__", None)
+    try:
+        version = metadata_version(f"{module}")
```

Comment:
The `str` of a module is not something you can pass to [`importlib.metadata.version`](https://docs.python.org/3/library/importlib.metadata.html#distribution-versions) because you get a repr-like string:

```python
>>> module = importlib.import_module("towncrier")
>>> print(f"{module}")
<module 'towncrier' from '/Users/.../site-packages/towncrier/__init__.py'>
```

which is probably just a small mistake here, but it reminds me that in fact `version()` does not accept an [Import Package](https://packaging.python.org/en/latest/glossary/#term-Import-Package) name at all, but rather a [Distribution Package](https://packaging.python.org/en/latest/glossary/#term-Distribution-Package). They're commonly the same but not necessarily so, to use some popular examples:

| Import Package | Distribution Package |
|----------------|----------------------|
| attr[^1]       | attrs                |
| yaml           | PyYAML               |

[^1]: attrs also has its newer `attrs` namespace of course :-)

As an example, let's say I'm the PyYAML developer, and I'm using towncrier with a `__version__` variable via `package = "yaml"`. If I want to drop that variable and specify my version in `pyproject.toml` instead, then we can't pass `yaml` to `importlib.metadata.version`. We have to load a mapping from one to the other:

```python
>>> import importlib.metadata
>>> importlib.metadata.packages_distributions()['yaml']
['PyYAML']
>>> importlib.metadata.version('PyYAML')
'6.0.1'
```

which in the context of `get_version` would be something like:

```python
dist_packages = distributions_packages()
try:
    dist_package = dist_packages[package]
except KeyError:
    # do fallback to package.__version__ here
else:
    if len(dist_package) > 1:
        raise Exception("I found more than one package")
    version = metadata_version(dist_package[0])
```

So that's how we'd make towncrier use importlib.metadata in a backwards-compatible way (user doesn't have to change their `tool.towncrier` config).

It's a *little* unfortunate, though, that I have to do this:

```toml
[project]
name = "PyYAML"
version = "1.2.3"

[tool.towncrier]
package = "yaml"
```

The version associated with the distribution package is right there, but then I have to tell towncrier one of the import packages I have so that it can look up the name of my distribution package.

PEP 621 says that the `project.name` is required to be statically defined, which means towncrier could try to read `project.name` by default if `package` is not defined and `--version` is not passed.

Notably Poetry doesn't yet support PEP 621 but the `package` would still be there for Poetry users.